src/NServiceBus.RabbitMQ.AcceptanceTests/ConfigureEndpointRabbitMQTransport.cs

```diff
@@ -76,13 +70,31 @@ ConnectionFactory CreateConnectionFactory()
     {
         var connectionFactory = new ConnectionFactory
         {
-            UserName = connectionStringBuilder["username"].ToString(),
-            Password = connectionStringBuilder["password"].ToString(),
-            VirtualHost = connectionStringBuilder["virtualhost"].ToString(),
-            HostName = connectionStringBuilder["host"].ToString(),
             AutomaticRecoveryEnabled = true
         };
 
+        object value;
+
+        if (connectionStringBuilder.TryGetValue("username", out value))
+        {
+            connectionFactory.UserName = value?.ToString();
```

Comment:
If the value is missing from the connection string, then `connectionStringBuilder.TryGetValue` will return `false` so this statement will never execute. Note that we have to use `TryGetValue` since `Get`, or the indexer will throw an exception if the value is not present.